File: api/internal/models/metrics.go
package models

import "time"

type AdminMetricsSummary struct {
	TotalCurrentInventoryItems int     `json:"total_current_inventory_items"`
	TotalCurrentApproxValue    float64 `json:"total_current_approx_value"`
	ForSaleCount               int     `json:"for_sale_count"`
	InUseCount                 int     `json:"in_use_count"`
	SalePendingCount           int     `json:"sale_pending_count"`
	SoldCount                  int     `json:"sold_count"`
	DonatedCount               int     `json:"donated_count"`
	DisposedCount              int     `json:"disposed_count"`
	ArchivedCount              int     `json:"archived_count"`
	AIEnrichedCount            int     `json:"ai_enriched_count"`
	MissingApproxValueCount    int     `json:"missing_approx_value_count"`
}

type AdminMetricsTopValueItem struct {
	ID              string   `json:"id"`
	Title           *string  `json:"title"`
	ApproxValue     *float64 `json:"approx_value"`
	DispositionCode *string  `json:"disposition_code"`
	ContainerID     *string  `json:"container_id"`
	ContainerName   *string  `json:"container_name"`
	PrimaryImageID  *string  `json:"primary_image_id"`
}

type AdminMetricsDuplicateItem struct {
	ID              string   `json:"id"`
	Title           *string  `json:"title"`
	ApproxValue     *float64 `json:"approx_value"`
	DispositionCode *string  `json:"disposition_code"`
	ContainerID     *string  `json:"container_id"`
	ContainerName   *string  `json:"container_name"`
}

type AdminMetricsDuplicateTitleGroup struct {
	NormalizedTitle  string                      `json:"normalized_title"`
	Count            int                         `json:"count"`
	TotalApproxValue float64                     `json:"total_approx_value"`
	Items            []AdminMetricsDuplicateItem `json:"items"`
}

type AdminMetricsResponse struct {
	Summary              AdminMetricsSummary               `json:"summary"`
	TopValueItems        []AdminMetricsTopValueItem        `json:"top_value_items"`
	DuplicateTitleGroups []AdminMetricsDuplicateTitleGroup `json:"duplicate_title_groups"`
	GeneratedDatetime    time.Time                         `json:"generated_datetime"`
}

File: frontend/src/types/metrics.ts
export interface AdminMetricsSummary {
  total_current_inventory_items: number;
  total_current_approx_value: number;
  for_sale_count: number;
  in_use_count: number;
  sale_pending_count: number;
  sold_count: number;
  donated_count: number;
  disposed_count: number;
  archived_count: number;
  ai_enriched_count: number;
  missing_approx_value_count: number;
}

export interface AdminMetricsTopValueItem {
  id: string;
  title: string | null;
  approx_value: number | null;
  disposition_code: string | null;
  container_id: string | null;
  container_name: string | null;
  primary_image_id: string | null;
}

export interface AdminMetricsDuplicateItem {
  id: string;
  title: string | null;
  approx_value: number | null;
  disposition_code: string | null;
  container_id: string | null;
  container_name: string | null;
}

export interface AdminMetricsDuplicateTitleGroup {
  normalized_title: string;
  count: number;
  total_approx_value: number;
  items: AdminMetricsDuplicateItem[];
}

export interface AdminMetricsResponse {
  summary: AdminMetricsSummary;
  top_value_items: AdminMetricsTopValueItem[];
  duplicate_title_groups: AdminMetricsDuplicateTitleGroup[];
  generated_datetime: string;
}
